### PR TITLE
feat: matchCurrentValue

### LINKED ISSUE #17950: Never try to update a dep with an explicit `=` range
### What would you like Renovate to be able to do?

I'd like any dep marked with `=` to never, ever be updated - the explicit equals sign means it's pinned for a reason.

### If you have any ideas on how this should be implemented, please tell us here.

Just automatically ignore any part of a range that has an explicit `=` sign.

### Is this a feature you are interested in implementing yourself?

Maybe

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Adds new packageRules.matchCurrentValue

## Context

Closes #17950

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/package-rules/current-value.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    { currentValue }: PackageRuleInputConfig,
+    { matchCurrentValue }: PackageRule
+  ): boolean | null {
+    if (is.undefined(matchCurrentValue)) {
+      return null;
+    }
+    const matchCurrentValueStr = matchCurrentValue.toString();
+    const matchCurrentValuePred = configRegexPredicate(matchCurrentValueStr);
+
+    if (!matchCurrentValuePred) {
```
Comment (rarkins) [inline lib/util/package-rules/current-value.ts:18]: hopefully shouldn't happen due to validation

File: lib/util/package-rules/current-value.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+export class CurrentValueMatcher extends Matcher {
+  override matches(
+    { currentValue }: PackageRuleInputConfig,
+    { matchCurrentValue }: PackageRule
+  ): boolean | null {
+    if (is.undefined(matchCurrentValue)) {
+      return null;
+    }
+    const matchCurrentValueStr = matchCurrentValue.toString();
```
[the remark this answers is not in the document]
Comment (rarkins) [reply lib/util/package-rules/current-value.ts:15]: image

[the remark this answers is not in the document]
Comment (rarkins) [reply lib/util/package-rules/current-value.ts:15]: Not sure why/how, it's the same for matchCurrentVersion

[the remark this answers is not in the document]
Comment (rarkins) [reply lib/util/package-rules/current-value.ts:15]: Seeing if I can fix for this one at least

**Review by viceice — COMMENTED** [review]
needs coverage

File: lib/util/package-rules/current-value.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      return null;
+    }
+    const matchCurrentValuePred = configRegexPredicate(matchCurrentValue);
+
+    if (!matchCurrentValuePred) {
+      logger.warn(
+        { matchCurrentValue },
+        'matchCurrentValue should be a regex, starting and ending with `/`'
+      );
+      return null;
```
Comment (secustor) [inline lib/util/package-rules/current-value.ts:22]: Should this really be handled as there is no rule at all?     
I think `false` would be better here, if somehow the validation fails the rule should not be applied.

**Review by secustor — COMMENTED** [review]
The Matcher has to be added to the central list to be picked up.    
https://github.com/renovatebot/renovate/blob/main/lib/util/package-rules/matchers.ts#L35

Outside of that LGTM.

File: lib/util/package-rules/current-value.ts (hunk 3 above)
Comment (rarkins) [reply lib/util/package-rules/current-value.ts:22]: Neither case is great, but I agree with you so have switched

**Review by viceice — APPROVED** [review]

**Review by secustor — APPROVED** [review]

**Pull request merged**